gazebo_plugins/include/gazebo_plugins/gazebo_ros_openni_kinect.h

```diff
@@ -14,134 +14,24 @@
  * limitations under the License.
  *
 */
-/*
- * Desc: A dynamic controller plugin that publishes ROS image_raw camera_info topic for generic camera sensor.
- * Author: John Hsu
- * Date: 24 Sept 2008
+
+/* DEPRECATED.
+ * Instead use, GazeboRosDepthCamera
  */
 
 #ifndef GAZEBO_ROS_OPENNI_KINECT_HH
 #define GAZEBO_ROS_OPENNI_KINECT_HH
 
-// ros stuff
-#include <ros/ros.h>
-#include <ros/callback_queue.h>
-#include <ros/advertise_options.h>
-
-// ros messages stuff
-#include <sensor_msgs/PointCloud2.h>
-#include <sensor_msgs/Image.h>
-#include <sensor_msgs/CameraInfo.h>
-#include <sensor_msgs/fill_image.h>
-#include <std_msgs/Float64.h>
-#include <image_transport/image_transport.h>
-
-// gazebo stuff
-#include <sdf/Param.hh>
-#include <gazebo/physics/physics.hh>
-#include <gazebo/transport/TransportTypes.hh>
-#include <gazebo/msgs/MessageTypes.hh>
-#include <gazebo/common/Time.hh>
-#include <gazebo/sensors/SensorTypes.hh>
-#include <gazebo/plugins/DepthCameraPlugin.hh>
-
-// dynamic reconfigure stuff
-#include <gazebo_plugins/GazeboRosOpenniKinectConfig.h>
-#include <dynamic_reconfigure/server.h>
-
-// boost stuff
-#include <boost/thread/mutex.hpp>
-
-// camera stuff
-#include <gazebo_plugins/gazebo_ros_camera_utils.h>
+// Base plugin
+#include <gazebo_plugins/gazebo_ros_depth_camera.h>
 
 namespace gazebo
 {
-  class GazeboRosOpenniKinect : public DepthCameraPlugin, GazeboRosCameraUtils
+  class GazeboRosOpenniKinect : public GazeboRosDepthCamera
```

Comment:
This looks like a big change to make in a release branch since it should break API.